src/iotEdgeExplorer.ts

```diff
@@ -102,6 +114,23 @@ export class IoTEdgeExplorer extends BaseExplorer {
         }
     }
 
+    private async isValidDeploymentJson(json: string): Promise<boolean> {
+        try {
+            const schema = (await axios.get(Constants.DeploymentJsonSchemaUrl)).data;
+            const ajv = new Ajv({ allErrors: true });
+            const valid = ajv.validate(schema, JSON.parse(json));
+            if (!valid) {
+                this._outputChannel.show();
+                this.outputLine(Constants.IoTHubEdgeLabel, `There are errors in deployment json file:\n${ajv.errorsText(null, { separator: ",\n" })}`);
+                return false;
+            }
+            return true;
+        } catch (error) {
+            vscode.window.showWarningMessage(`Cannot validate deployment json: ${error.toString()}`);

```

Comment:
the message here maybe annoying. User could have correct schema, but met with this warning. Do we really need user aware of it here? Or we just need collect some telemetry data?

Reply:
Only send telemetry instead